### PR TITLE
[KYUUBI #3406] [Subtask] [Doc] Add PySpark client docs

### LINKED ISSUE #3406: [Subtask] [Doc] Add PySpark client docs
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

Add PySpark client docs

### Parent issue

#3039

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

close #3406.

Add PySpark client docs.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Thanks for adding this document, have some high-level thoughts.

We should not narrow the Kyuubi JDBC data source use cases into PySpark, it's also suitable for `spark-shell`, `spark-sql` and the normal spark jar jobs, right? So, I think maybe we can split the doc into two parts, the 1st part is "How to use Kyuubi as a Spark JDBC data source", and the 2nd is "How to use PySpark to access Kyuubi JDBC data source".

**Comment by pan3793** [comment]
Also, to unconfusing users, we should emphasize there are TWO Spark applications in such use architects, and the most benefit is security but not performance, the arch also suitable for `spark-shell`

**Comment by bowenliang123** [comment]
Yes. But so far this docs is purposed for python user (like AI team and etc.) accessing to Kyuubi and it is put under `/client/python` rather than discussing all avaialble accessing way in Spark which could be enhanced in future docs solely for Spark under `/client`.  And they are not confusing with two spark jobs as discussed in first paragraph it is about connect to Kyuubi but not the any realted to Spark engine.

File: docs/client/python/pyspark.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+| Driver     | Driver Class Name | Remarks|
+| ---------- | ----------------- | ----- |
+| Kyuubi Hive Driver ([doc](../jdbc/kyuubi_jdbc.html))| org.apache.kyuubi.jdbc.KyuubiHiveDriver |  Compile for the driver on master branch, as [KYUUBI #3484](https://github.com/apache/incubator-kyuubi/pull/3485) required by Spark JDBC source not yet included in released version.
+| Hive Driver ([doc](../jdbc/hive_jdbc.html))| org.apache.hive.jdbc.HiveDriver | 
+
+Refer to docs of the driver and prepare the JDBC driver jar file.
+
+### Prepare JDBC Hive Dialect extension
+
+Hive Dialect support is requried by Spark for wraping sql correctly and sending to JDBC driver. Kyuubi provides a JDBC dialect extension with auto regiested Hive Daliect support for Spark. Follow the instrunctions in [Hive Dialect Support](../../engines/spark/jdbc-dialect.html) to prepare the plugin jar file `kyuubi-extension-spark-jdbc-dialect_-*.jar`.
```
Comment (pan3793) [inline docs/client/python/pyspark.md:49]: sql => SQL

File: docs/client/python/pyspark.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+Choose one of following ways to include jar files to Spark.
+
+- Put the jar file of JDBC driver and Hive Dialect to `$SPARK_HOME/jars` directory to make it visible for the classpath of PySpark. And adding `spark.sql.extensions = org.apache.spark.sql.dialect.KyuubiSparkJdbcDialectExtension` to `$SPARK_HOME/conf/spark_defaults.conf.`
+
+- With spark's start shell, include JDBC driver when you submit the application with `--packages`, and the Hive Dialect plugins with `--jars`
+
+```
+$SPARK_HOME/bin/pyspark --py-files PY_FILES \
+--packages org.apache.hive:hive-jdbc:x.y.z \
```
Comment (pan3793) [inline docs/client/python/pyspark.md:61]: Can we use kyuubi hive driver in demo code?

File: docs/client/python/pyspark.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+For further information about PySpark JDBC usage and options, please refer to Spark's [JDBC To Other Databases](https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html).
+
+### Reading and Writing via JDBC data source
+
+```python
+# Loading data from Kyuubi via HiveDriver as JDBC source
+jdbcDF = spark.read \
+  .format("jdbc") \
+  .options(driver="org.apache.hive.jdbc.HiveDriver",
```
Comment (pan3793) [inline docs/client/python/pyspark.md:88]: ditto, prefer to use KyuubiHiveDriver

File: docs/client/python/pyspark.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+           dbtable="testdb.tgt_table"
+           ) \
+    .save()
+```
+
+
+### Use PySpark with Pandas
+From PySpark 3.2.0, PySpark supports pandas API on Spark which allows you to scale your pandas workload out.
+
+Pandas-on-Spark DataFrame and Spark DataFrame are virtually interchangeable.More instructions in [From/to pandas and PySpark DataFrames](https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/pandas_pyspark.html#pyspark).
```
Comment (pan3793) [inline docs/client/python/pyspark.md:113]: quote interchangeable.More by `

**Review by pan3793 — APPROVED** [review]
Thanks, only minor comments

File: docs/client/python/pyspark.md (hunk 2 above)
Comment (bowenliang123) [reply docs/client/python/pyspark.md:61]: Negative for right now. As `isSigned` implementation of KyuubiHiveDriver #3485 not yet released in latest 1.6.0 release.

File: docs/client/python/pyspark.md (hunk 3 above)
Comment (bowenliang123) [reply docs/client/python/pyspark.md:88]: same obstacle as above.

File: docs/client/python/pyspark.md (hunk 4 above)
Comment (bowenliang123) [reply docs/client/python/pyspark.md:113]: sorry for Misdirection. A blank is now added before `More`.

File: docs/client/python/pyspark.md (hunk 2 above)
Comment (pan3793) [reply docs/client/python/pyspark.md:61]: Emm... 1.6.1 is scheduled mid-Oct. then let's update doc after that.

**Comment by pan3793** [comment]
Thanks, merging to master/1.6

**Pull request closed without merging**